### ISSUE TITLE
Aggregation doesn't give usable/useful results when marked entities overlap

### ISSUE THREAD

**Issue by alexbfree** [zooniverse/aggregation#144]
For example, here are a couple of cases from Snapshot Serengeti: Computer Vision's recent beta

[Subject 1925326](https://www.zooniverse.org/projects/alexbfree/snapshot-serengeti-computer-vision/talk/subjects/1925326) (formerly [ASG0018whg](https://talk.snapshotserengeti.org/#/subjects/ASG0018whg) in Serengeti)

The crowd opinions are:
![asg0018whg](https://cloud.githubusercontent.com/assets/1473244/14850266/c55fc494-0c6f-11e6-819b-0db28d457753.jpg)
(collectively there are 55 annotations)

but this aggregates to 
![asg0018whg](https://cloud.githubusercontent.com/assets/1473244/14850271/cf9d71a4-0c6f-11e6-9595-17fc4ce2d68d.jpg)
which still has 17 clusters!

Compare this to a better example, subject [1714356](https://www.zooniverse.org/projects/alexbfree/snapshot-serengeti-computer-vision/talk/subjects/1714356) (formerly [ASG001c9b4](https://talk.snapshotserengeti.org/#/subjects/ASG001c9b4)):
![asg001c9b4](https://cloud.githubusercontent.com/assets/1473244/14850332/38290116-0c70-11e6-89a0-2f466a51280c.jpg)
(57 clusters)
which aggregates to 
![asg001c9b4](https://cloud.githubusercontent.com/assets/1473244/14850363/6fb503f0-0c70-11e6-84a9-4c57d938cba1.jpg)
which has 2 clusters, exactly corresponding to the number of animals in the picture - Perfect! Very usable and exactly what we want from the output.

In the case of this project, the science team want to use the aggregate answer to determine what images to crop out and use for training the animal detection computer vision system.
Answers like the first one can't be used (and also, can't easily be pulled out and separated from the "good" answers even if we did find some way to handle it manually)

I realise that overlapping animals/regions is a hard problem, but right now the code seems to give up and it doesn't even mark that that image has been handled differently.

Also, there are some cases where the aggregation seems to do a poor job and lose important data.
Here is another example:

[Subject 1925354](https://www.zooniverse.org/projects/alexbfree/snapshot-serengeti-computer-vision/talk/subjects/1925354) (formerly [ASG001e0wq](https://talk.snapshotserengeti.org/#/subjects/ASG001e0wq))
The crowd opinions look like this:
![asg001e0wq](https://cloud.githubusercontent.com/assets/1473244/14850505/7b49fa12-0c71-11e6-8a5a-ddf9589de718.jpg)
It's quite clear that there almost everyone in the crowd agrees on the approximate location of the three zebras, and their three different directions.
The result from the aggregation is, in light of this, not at all useful, and missing important detail on the presence of three animals and their position/direction. Note that the left and right animals are lost, and only a middle animal is shown (wrongly):
![asg001e0wq](https://cloud.githubusercontent.com/assets/1473244/14850531/a774bbc2-0c71-11e6-9f64-6bb61eef5ae8.jpg)

In summary, I think overall we need to
1. Improve aggregation answers for overlapping entities
2. Where a good answer is not easy/possible, add some tagging or metadata to that image's results to show that it is suboptimal/needs further attention.

The second is most important of all, I think.

BTW for a sense of the spread of this problem, 23 images of the 66 from the beta are subject to these issues -> about a third. We definitely need to address this if aggregation is to be useful.

**Comment by ggdhines-zz** [comment]
The rectangle marking tool wasn't designed to handle this sort of case. Perfectly reasonable case to want to handle and I don't think it should be too hard. Semantics but I would count this as an enhancement and not a bug. (I can think of projects where the above behaviour would be wanted - for example using rectangles to outline grass.) So I'll need to figure out a way to report both types of results

**Comment by alexbfree** [comment]
OK, I take your point that it could be deemed an enhancement to deal with situations like this - but can we at least tag the outputs to identify which images are successfully combined and which aren't? I would say without that, the aggregation is not usable for a project like this.

**Comment by ggdhines-zz** [comment]
There is always a chance of error. Not just for rectangle aggregation but for all aggregation algorithms and citizen science in general. So if you need 100% accuracy, there's not much I can do for you. The closest example I can think of is Penguin Watch. The clustering there is pretty accurate but sometimes makes mistakes - the only way to know for sure is to examine some of the images.

**Comment by alexbfree** [comment]
I'm not saying we need to eliminate error. I'm saying we need to differentiate (through some metadata tag or extra field), the "ones we can't do", from the "ones we can".... Seems the algorithm is succeeding in some cases, and failing in others, so if the algorithm knows the difference, we should export that fact in the CSVs

**Comment by alexbfree** [comment]
Discussed with Greg and he pointed out it is difficult to deduce this algorithmically.

One idea (for users of export data to discern good from bad, or for a heuristic/recommendation the export could provide):
One thing we do have in the aggregate output is the average number of clusters marked. So it should be possible to filter subjects based on that... if the number of rectangles post-aggregation is vastly more than the average number of clusters marked per user then we know further attention is definitely needed. If it’s the same or +/- 2, then maybe the results are more likely to be useful.....

**Comment by chrislintott** [comment]
It’s not that hard - restricting clusters to one per user should do it. See Andromeda Project data reductions for example
